### PR TITLE
[C++] Introduce userver

### PR DESCRIPTION
**Pull request by itrofimow**
This PR aims to incorporate https://github.com/userver-framework/userver into benchmark suite

### CODE REVIEW

**Comment by itrofimow** [comment]
@apolukhin @userver-framework can you guys please have a look at this?

There def. are some configurations tweaks to make, but i'm not yet certain which ones exactly - guess it can be done after merging this PR and seeing benchmark results.

**Comment by itrofimow** [comment]
> A few nitpicks for later changes:
> 
> * Please write userver in lower case
> * You could replace userver::formats::json::ValueBuilder with formats::json::StringBuilder to slightly speedup the service
> * Adjusting ev and worker threads count would improve the results. Experiment qith those valies to find the best ones.
> * If target architecture is know, use -march= and -mtune= compiler flags to save a few more CPU cycles

Fixed userver spelling, added `march=native`. 

- About `formats::json::StringBuilder` - quoting TFB docs, "All test implementations should be production-grade", and i'm not sure how production-grade that is (but feel free to disagree and fix it later).

- `Adjusting ev and worker threads count`: that definitely should be done, but i feel like it's best changed with some results at hand

**Comment by itrofimow** [comment]
@nbrady-techempower Hi! Could you please have a look at this?

**Comment by apolukhin** [comment]
> About formats::json::StringBuilder - quoting TFB docs, "All test implementations should be production-grade", and i'm not sure how production-grade that is (but feel free to disagree and fix it later).

We use it quite a lot in production for serializing responses. It's an interface for streaming serializer (SAX), which is a quite a common technique. There's also a separate section in docs with a sample for SAX streaming https://userver.tech/db/dd7/md_en_userver_formats.html#streamingserialization . However, it may not provide notable advantage for small responses.

File: frameworks/C++/userver/userver_benchmark/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        pg_->Execute(userver::storages::postgres::ClusterHostType::kMaster,
+                     "SELECT id, randomNumber from World where id = $1",
+                     userver::utils::RandRange(1, kMaxWorldRows + 1))
+            .AsSingleRow<TableRow>(userver::storages::postgres::kRowTag);
+
+    userver::formats::json::ValueBuilder builder{
+        userver::formats::json::Type::kObject};
+    builder["id"] = row.id;
+    builder["randomNumber"] = row.random_number;
+    return builder.ExtractValue();
```
Comment (nUl1) [inline frameworks/C++/userver/userver_benchmark/src/main.cpp:86]: an inline builder (`formats::json::MakeDoc`) should perform better here

**Comment by itrofimow** [comment]
Not sure what went wrong here, runs fine in my action queue: https://github.com/itrofimow/FrameworkBenchmarks/pull/1

@nbrady-techempower could you please rerun failing checks for me?

**Comment by itrofimow** [comment]
Hi @nbrady-techempower! 
Thank you for rerunning the checks.

Is there anything holding this PR back from being merged?

**Pull request merged**